src/ray/raylet/worker_pool.cc

```diff
@@ -38,6 +38,28 @@ bool RemoveWorker(std::unordered_set<std::shared_ptr<ray::raylet::Worker>> &work
   return worker_pool.erase(worker) > 0;
 }
 
+// A helper function to remove a worker client from a map of pid to list of clients.
+// If the list of clients is empty after removing this one, erases it from the map.
+bool RemoveConnection(
+    std::unordered_map<pid_t, std::vector<std::shared_ptr<ray::LocalClientConnection>>>
```

Comment:
Note that this will copy the whole unordered map, which is not a super big deal, since it will generally not have more than a few 100 elements and this function is not in the critical path, but you should consider passing it as a const reference.